### PR TITLE
Issue33 math func

### PR DESCRIPTION
**Pull request by CanBerker**
All mathematical functions have been implemented.

When implementing "round" and "roundhalftoeven" I defined a rounding function in the "RoundFunctionIterator" which is used by both iterators. If you have suggestions for a better practise or location to define the function please let me know.

With the recently added functions the functions folder is getting quite crowded. Moving them to a "Numeric" folder or multiple folders such as "trigonometric", "exponential" and so on can be beneficial for organization.

### CODE REVIEW

File: src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/RoundFunctionIterator.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        } else
+            throw new IteratorFlowException(RuntimeIterator.FLOW_EXCEPTION_MESSAGE + " round function", getMetadata());
+    }
+
+
+    /**
+     * Decimal Formatting Solution
+     * https://stackoverflow.com/questions/153724/how-to-round-a-number-to-n-decimal-places-in-java
+     */
+    public static Double getRoundedResult(Double value, Integer precision, RoundingMode rm) {
```
Comment (wscsprint3r) [inline src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/RoundFunctionIterator.java]: This needs to be moved to some utility class, it doesn't make sense to have it in the Iterator as a static method. 

It would be good if you could find a pre-existing solution since rounding of doubles is not an uncommon problem. 
This may or may not help you: 
https://commons.apache.org/proper/commons-math/javadocs/api-3.3/org/apache/commons/math3/util/Precision.html

File: src/main/resources/test_files/runtime/FunctionRoundHalfToEven.iq [hunk 2]
```diff
@@ -0,0 +1,7 @@
+(:JIQS: ShouldRun; Output="( 0.0, 2.0, 2.0, 3567.81, 0.0, 35600.0)" :)
+roundhalftoeven(0.5),
```
Comment (ghislainfourny) [inline src/main/resources/test_files/runtime/FunctionRoundHalfToEven.iq]: the name should be round-half-to-even

File: src/main/resources/test_files/runtime/FunctionSqrt.iq [hunk 3]
```diff
@@ -0,0 +1,6 @@
+(:JIQS: ShouldRun; Output="( 0.0, -0.0, 1000.0)" :)
+sqrt(0.0e0),
```
Comment (ghislainfourny) [inline src/main/resources/test_files/runtime/FunctionSqrt.iq]: It would be great to add a test for sqrt of a negative number (double-checking which error should be returned in the specification)

**Comment by ghislainfourny** [comment]
This is awesome work, Can. I have added my minor comments inline.
